frontend/pulse3d/src/components/interactiveAnalysis/InteractiveAnalysisModal.js

```diff
@@ -271,20 +271,17 @@ export default function InteractiveWaveformModal({
 
   const checkDuplicates = (well) => {
     const wellToUse = well ? well : selectedWell;
+    const { startTime, endTime } = editableStartEndTimes;
 
     let peaksList = editablePeaksValleys[wellToUse][0].sort((a, b) => a - b);
     let valleysList = editablePeaksValleys[wellToUse][1].sort((a, b) => a - b);
 
-    // if data for a well was never fetched, assume no filtering required because the user would not have moved the peaks and valley lines. Just checked duplicates against original peaks and valleys
-    if (wellToUse in originalData.coordinates) {
-      const dataToCompare = originalData.coordinates[wellToUse];
-
-      peaksList = peaksList.filter((peak) => dataToCompare[peak][1] >= peakValleyWindows[wellToUse].minPeaks);
-      valleysList = valleysList.filter(
-        (valley) => dataToCompare[valley][1] <= peakValleyWindows[wellToUse].maxValleys
-      );
-    }
-
+    // Filter before looking for duplicates
+    const wellCoords = originalData.coordinates[wellToUse]
```

Comment:
In general for JS, if you see the pattern `c = a ? a : b;` you can reduce it to `c = a || b;`